rocAL/rocAL/source/pipeline/master_graph.cpp

```diff
@@ -796,13 +796,78 @@ MasterGraph::copy_out_tensor(void *out_ptr, RocalTensorFormat format, float mult
                     else if(output_data_type == RocalTensorDataType::FP16) {
                         half *output_tensor_16 = static_cast<half *>(out_ptr);
                         auto channel_size = w * h;
+    #if (ENABLE_SIMD && __AVX2__)
+                        half *B_buf = output_tensor_16 + dest_buf_offset;
+                        half *G_buf = B_buf + channel_size;
+                        half *R_buf = G_buf + channel_size;
+
+                        __m256i mask_B, mask_G, mask_R;
+                        if (reverse_channels) {
+                            mask_B = _mm256_setr_epi32(0x80808000, 0x80808003, 0x80808006, 0x80808009, 0x80808000,
+                                                        0x80808003, 0x80808006, 0x80808009);
+                            mask_G = _mm256_setr_epi32(0x80808001, 0x80808004, 0x80808007, 0x8080800A, 0x80808001,
+                                                        0x80808004, 0x80808007, 0x8080800A);
+                            mask_R = _mm256_setr_epi32(0x80808002, 0x80808005, 0x80808008, 0x8080800B, 0x80808002,
+                                                        0x80808005, 0x80808008, 0x8080800B);
+                        } else {
+                            mask_R = _mm256_setr_epi32(0x80808000, 0x80808003, 0x80808006, 0x80808009, 0x80808000,
+                                                        0x80808003, 0x80808006, 0x80808009);
+                            mask_G = _mm256_setr_epi32(0x80808001, 0x80808004, 0x80808007, 0x8080800A, 0x80808001,
+                                                        0x80808004, 0x80808007, 0x8080800A);
+                            mask_B = _mm256_setr_epi32(0x80808002, 0x80808005, 0x80808008, 0x8080800B, 0x80808002,
+                                                        0x80808005, 0x80808008, 0x8080800B);
+                        }
+                        __m256 pmul0 = _mm256_set1_ps(multiplier0);
+                        __m256 pmul1 = _mm256_set1_ps(multiplier1);
+                        __m256 pmul2 = _mm256_set1_ps(multiplier2);
+                        __m256 padd0 = _mm256_set1_ps(offset0);
+                        __m256 padd1 = _mm256_set1_ps(offset1);
+                        __m256 padd2 = _mm256_set1_ps(offset2);
+                        unsigned int alignedLength = (channel_size & ~7);    // multiple of 8
+                        unsigned int i = 0;
+
+                        __m256 fR, fG, fB;
+                        float dstPtrTempR_ps[8], dstPtrTempG_ps[8], dstPtrTempB_ps[8];
```

Comment:
Rename 
dstPtrTempR_ps[8], dstPtrTempG_ps[8], dstPtrTempB_ps[8];
as 
TempR[8], TempG[8]. TempB[8]

Reply:
Resolved in latest commit